openjdk6-b22/jdk/src/share/classes/javax/swing/plaf/basic/BasicComboPopup.java

```diff
@@ -1232,11 +1232,9 @@ protected Rectangle computePopupBounds(int px,int py,int pw,int ph) {
             screenBounds = new Rectangle(p, toolkit.getScreenSize());
         }
 
+        if (ph > screenBounds.height && screenBounds.height > 500)
```

Comment:
Does it exist screen with less than 500 pixels ? :-)

Reply:
The code above can return negative screen height. So this check is required. It can be check >100 or >50 but 500 is reasonable for modern PC.